src/shared/varlink-io.systemd.Resolve.c

```diff
@@ -171,17 +172,20 @@ static SD_VARLINK_DEFINE_ERROR(BadAddressSize);
 static SD_VARLINK_DEFINE_ERROR(ResourceRecordTypeInvalidForQuery);
 static SD_VARLINK_DEFINE_ERROR(ZoneTransfersNotPermitted);
 static SD_VARLINK_DEFINE_ERROR(ResourceRecordTypeObsolete);
+static SD_VARLINK_DEFINE_ERROR(InconsistentServiceRecords);
 
 SD_VARLINK_DEFINE_INTERFACE(
                 io_systemd_Resolve,
                 "io.systemd.Resolve",
                 &vl_method_ResolveHostname,
                 &vl_method_ResolveAddress,
+                SD_VARLINK_SYMBOL_COMMENT("Resolves a named DNS-SD or unnamed simple SRV service."),
                 &vl_method_ResolveService,
                 &vl_method_ResolveRecord,
                 &vl_type_ResolvedAddress,
                 &vl_type_ResolvedName,
                 &vl_type_ResolvedService,
+                SD_VARLINK_SYMBOL_COMMENT("Encodes the canonical name, type and domain of a DNS-SD or simple SRV service. Note that due to CNAME redirections and similar, a named DNS-SD service might resolve to a canonical service that is an unnamed simple SRV service. Or in other words: resolving a named service might return an unnamed canonical service."),
```

Comment:
Haha, I think we should not return such spurious result. But, that's out of the scope of this PR, and already filed as #33947. Let's fix the issue later.